### PR TITLE
Create `ws_protocol_cls` fixture

### PR DESCRIPTION
**Pull request by Kludex**
This PR aims to remove the `WS_PROTOCOLS` "constant" that we have on the `test_websockets.py`.

### CODE REVIEW

File: tests/middleware/test_proxy_headers.py [hunk 1]
```diff
@@ -1,24 +1,19 @@
...
-
-try:
-    import websockets.client
 
+if TYPE_CHECKING:
     from uvicorn.protocols.websockets.websockets_impl import WebSocketProtocol
-
-    WS_PROTOCOLS = [WSProtocol, WebSocketProtocol]
-except ImportError:  # pragma: nocover
-    WS_PROTOCOLS = []
```
Comment (Kludex) [inline tests/middleware/test_proxy_headers.py:21]: This was incorrect. The `WebSocketProtocol` will always be available.

File: tests/supervisors/test_reload.py
Comment (Kludex) [inline tests/supervisors/test_reload.py:1]: I need to create an issue to investigate those. They are failing locally quite often.

EDIT: I'm usually using an M2.

**Review by Kludex — COMMENTED** [review]
I need to replace `type` by `Type`. 😢

**Review by iudeen — COMMENTED** [review]
This looks good!

File: tests/supervisors/test_reload.py [hunk 2]
```diff
@@ -232,7 +253,18 @@ def test_should_not_reload_when_only_subdirectory_is_watched(
...
         reloader.shutdown()
 
-    @pytest.mark.parametrize("reloader_class", [WatchFilesReload, WatchGodReload])
+    @pytest.mark.parametrize(
+        "reloader_class",
+        [
+            pytest.param(
+                WatchFilesReload,
+                marks=pytest.mark.skipif(
+                    sys.platform == "darwin", reason="Flaky on MacOS"
```
Comment (aminalaee) [inline tests/supervisors/test_reload.py]: Don't you want to narrow it down to only Apple Silicon architecture?
Maybe use platform module for that?

**Review by aminalaee — APPROVED** [review]

File: tests/supervisors/test_reload.py (hunk 2 above)
Comment (Kludex) [reply tests/supervisors/test_reload.py]: How do I do that?

Comment (aminalaee) [reply tests/supervisors/test_reload.py]: I think something along this to check both OS and CPU architecture:

```py
import platform

system = platform.system() # "Darwin"
processor = platform.processor() # "arm"

@pytest.mark.skipif(system == "Darwin" and processor == "arm", reason="Flakey on MacOS")
```

Comment (Kludex) [reply tests/supervisors/test_reload.py]: done

**Pull request merged**